### DISCUSSION TITLE
Collect IP address on table insert

### DISCUSSION THREAD

**Discussion by iftachhh** [supabase/supabase#8771] [Questions] [answered]
Hey there! I've been wondering if there's a way to collect IP addresses of users upon different actions as insert to table or update it, of course I can manually insert it from the client side but its not efficient. Thanks!

**Comment by GaryAustin1** [comment, marked as answer]
You can get headers for database API operations thru PostgREST settings.
https://postgrest.org/en/stable/api.html#http-context
`SELECT current_setting('request.headers', true)::json->>'x-real-ip';` for example.

These headers contain the IP on a Supabase cloud instance.  Note they are coming thru CloudFlare.
x-real-ip
x-forwarded-for

**Reply by iftachhh** [reply]
THANKS! I used it that way:

```javascript
export const supabase = createClient(supabaseUrl, supabaseAnonKey, {
  headers: {
    fingerprint // use fingerprintjs, clientjs, etc..
  },
});

```

```sql
if exists (select * from information_schema.columns where table_name = TG_TABLE_NAME and column_name = 'biometrics') then
    new.biometrics = json_build_object(
        'ip', split_part(current_setting('request.headers', true)::json->>'x-forwarded-for', ', ', 1),
        'user_agent', current_setting('request.headers', true)::json->>'user-agent',
        'fingerprint', current_setting('request.headers', true)::json->>'fingerprint'
    );
  end if;
```
